File: Leetcode/Daily/06-2025/C++/16.cpp
// Problem: 2016. Maximum Difference Between Increasing Elements (Easy)
// Link: https://leetcode.com/problems/maximum-difference-between-increasing-elements

class Solution {
public:
    int maximumDifference(vector<int>& nums) {
        int n = nums.size(), maxx = 0, ans = -1;

        for (int i = n-1; i >= 0; i--) {
            if (maxx > nums[i]) ans = max(ans, maxx - nums[i]);
            maxx = max(maxx, nums[i]);
        }

        return ans;
    }
};

File: Leetcode/Daily/06-2025/Python3/16.py
# Problem: 2016. Maximum Difference Between Increasing Elements (Easy)
# Link: https://leetcode.com/problems/maximum-difference-between-increasing-elements

class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        n, maxx, ans = len(nums), 0, -1

        for val in nums[::-1]:
            if maxx > val:
                ans = max(ans, maxx - val)
            
            maxx = max(maxx, val)

        return ans